tests/api/v2/handlers/test_operations_api.py

```diff
@@ -0,0 +1,241 @@
+
+    async def test_unauthorized_get_operations(self, api_client):
+        resp = await api_client.get('/api/v2/operations')
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_get_operation_by_id(self, api_client, api_cookies):
+        resp = await api_client.get('/api/v2/operations/123', cookies=api_cookies)
+        operation_dict = await resp.json()
+        assert operation_dict['name'] == 'My Test Operation'
+        assert operation_dict['id'] == '123'
+
+    async def test_unauthorized_get_operation_by_id(self, api_client):
+        resp = await api_client.get('/api/v2/operations/123')
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_get_operation_report(self, api_client, api_cookies, mocker, async_return):
+        with mocker.patch('app.objects.c_operation.Operation.all_facts') as mock_all_facts:
+            mock_all_facts.return_value = async_return([])
+            test_objective = Objective(id='123', name='test objective', description='test', goals=[])
+            BaseService.get_service('data_svc').ram['operations'][0].objective = test_objective
+            resp = await api_client.get('/api/v2/operations/123/report', cookies=api_cookies)
+            report = await resp.json()
+            assert report['name'] == 'My Test Operation'
+            assert report['jitter'] == '2/8'
+
+    async def test_unauthorized_get_operation_report(self, api_client):
+        resp = await api_client.get('/api/v2/operations/report')
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_create_operation(self, api_client, api_cookies):
+        payload = dict(name='post_test', planner={'id': '123'},
+                       adversary={'adversary_id': '123'}, source={'id': '123'})
+        resp = await api_client.post('/api/v2/operations', cookies=api_cookies, json=payload)
+        assert resp.status == HTTPStatus.OK
+        op_data = await resp.json()
+        assert op_data.get('name') == "post_test"
+        op_exists = await BaseService.get_service('data_svc').locate('operations', {'name': 'post_test'})
+        assert op_exists
+
+    async def test_unauthorized_create_operation(self, api_client):
+        payload = dict(name='post_test', planner={'id': '123'},
+                       adversary={'adversary_id': '123'}, source={'id': '123'})
+        resp = await api_client.post('/api/v2/operations', json=payload)
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_update_operation(self, api_client, api_cookies):
+        payload = dict(state='running', obfuscator='base64')
+        resp = await api_client.patch('/api/v2/operations/123', cookies=api_cookies, json=payload)
+        assert resp.status == HTTPStatus.OK
+        op = (await BaseService.get_service('data_svc').locate('operations', {'id': '123'}))[0]
+        assert op.state == payload['state']
+        assert op.obfuscator == payload['obfuscator']
+
```

Comment:
we should also test:
1) updating a nonexistent operation
2) trying to update fields that can't be updated